src/elementary/src/absorb.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef NJOY_ELEMENTARY_ABSORB
+#define NJOY_ELEMENTARY_ABSORB
+
+// system includes
+
+// other includes
+#include "elementary/IsotopeID.hpp"
+
+namespace njoy {
+namespace elementary {
+
+  /**
+   *  @brief Return the isotope identifier when a compound absorbs a particle
+   *
+   *  @param[in] compound   the compound identifier (fundamental particle,
+   *                        isotope, nucleus, nuclide or the general particle)
+   *  @param[in] particle   the absorbed particle identifier (fundamental
+   *                        particle, isotope, nucleus, nuclide or the general
+   *                        particle)
+   */
+  template < typename Compound, typename Particle >
+  IsotopeID absorb( const Compound& compound, const Particle& particle ) {
+
+    try {
+
+      return IsotopeID( compound.za() + particle.za() );

```

Comment:
Interesting idea here. I like it. Hopefully not too simplistic.

Reply:
This will be used in reaction and decay resolution. I used a similar approach in software when I was still working in France. I used an isotope to avoid doing this for particle identifiers. You can however construct nuclide and nucleus identifiers by using the resulting isotope and a separate level.